### ISSUE TITLE
"2" == 2 Correct?

### ISSUE THREAD

**Issue by Glidias** [inkle/ink#116]
```
=== test_null_string_int_compare
~temp numStrTest = "2"
~temp numTest = 2
~temp nullFakeStrTest = "null"
~temp zeroVal = 0
"2"==2: {numStrTest==numTest:Yes|No}
"null"==0: {nullFakeStrTest==zeroVal:Yes|No}
->DONE
```

Result:

```
"2"==2: Yes
"null"==0: No
```

It seems inkle will equate a matching string data-type and number data-type as "the same" . Is this normal/intended? I guess, from a non-programmer perspective, this appears more "friendly". Also, there's no NULL value right, so a workaround would be to use a fake "null" string convention value i guess?

**Comment by joethephish** [comment]
Yeah, this is intentional, and comes out of the type-coalescing functionality that also allows you to use other binary operations, such as `2 + 3.1`. I agree it's slightly surprising that it works for equality on integers and strings, but unless someone can think of a really good reason for why it should be explicitly prevented, I don't see any harm in it.

Nope, there's no null value. Do you mean for when there's an empty divert target? Jon uses:

```
VAR nextPlace = -> none

// Unused
== none ==
ERROR: It shouldn't be possible to get here
-> DONE
```

Though it's not very elegant, so perhaps it should be possible to have an explicit `-> NONE` or something.
